examples/connect/CMakeLists.txt

```diff
@@ -7,20 +7,28 @@
 
 # This is a very simple example of using SOCI in a CMake-based project.
 
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.23...3.31 FATAL_ERROR)
```

Comment:
Is 3.23 really needed even for the examples (which don't use `FILE_SET`)?

Reply:
No. It's just that old CMake versions are being deprecated by Kitware. I'll change the minimum to 3.10 as that's the oldest  non-deprecated release as of now.